content/news/2024-02-17-bevy-0.13/index.md

```diff
@@ -0,0 +1,3208 @@
+
+## Entity Optimizations
+
+<div class="release-feature-authors">authors: @Bluefinger, @notverymoe, @scottmcm, @bushrat011899, @james7132</div>
+
+`Entity` (Bevy's 64-bit unique identifier for enitities) received a number of changes this cycle, combining laying some more groundwork for relations alongside _related_, and nice to have, performance optimizations. The work here involved a lot of deep-diving into compiler codegen/assembly output, with running lots of benchmarks and testing in order to ensure all changes didn't cause breakages or major problems. Although the work here was dealing with mostly _safe_ code, there were lots of underlying assumptions being changed that could have impacted code elsewhere. This was the most "micro-optimization" oriented set of changes in Bevy 0.13.
+
+* [#9797]: created a unified identifier type, paving the path for us to use the same fast, complex code in both our `Entity` type and the much-awaited relations
+* [#9907]: allowed us to store `Option<Entity>` in the same number of bits as `Entity`, by changing the layout of our Entity type to reserve exactly one `u64` value for the `None` variant
+* [#10519]: swapped us to a manually crafted `PartialEq` and `Hash` implementation for `Entity` to improve speed and save instructions in our hot loops
+* [#10558]: combined the approach of [#9907] and [#10519] to optimize `Entity`'s layout further, and optimized our `PartialOrd` and `Ord` implementations!
+* [#10648]: further optimized our entity hashing, changing how we multiply inside of the hash to save one precious assembly instruction in the optimized compiler output
+
+Full credit is also due to the authors who pursued similar work in [#2372] and [#3788]: while their work was not ultimately merged, it was an incredibly valuable
+inspiration and source of prior art to base these more recent changes on.
+
+![Benchmark results of optimisation work](entity_hash_optimsation_benches.png)
+
+The above results show from where we started (`optimised_eq` being the first PR that introduced the benchmarks) to where we are now with all the optimisations in place (`optimised_entity`).
+There are improvements across the board, with clear performance benefits that should impact multiple areas of the codebase, not just when hashing entities.
+
+There are a ton of crunchy, well-explained details in the linked PRs, including some fascinating assembly output analysis.
+If that interests you, open some new tabs in the background!
+
+[#9797]: https://github.com/bevyengine/bevy/pull/9797
+[#9907]: https://github.com/bevyengine/bevy/pull/9907
+[#10519]: https://github.com/bevyengine/bevy/pull/10519
+[#10558]: https://github.com/bevyengine/bevy/pull/10558
+[#10648]: https://github.com/bevyengine/bevy/pull/10648
+[#2372]: https://github.com/bevyengine/bevy/pull/2372
+[#3788]: https://github.com/bevyengine/bevy/pull/3788
+
+## Porting `Query::for_each` to `QueryIter::fold` override
```

Comment:
This _was_ a sub-heading at one point, but was elevated, so maybe this needs the `<div class="release-feature-author">` part here as well.